### DISCUSSION TITLE
Dynamically setting a multiselect field options.

### DISCUSSION THREAD

**Discussion by IRodriguezSC** [sadieblue/agilityblue.workspace-scripts#34] [Q&A] [answered]
Upon creation of a task/form, is there a way to dynamically add the values available in a multi-select field.  For example, we are trying to load the multi-select field with specific billing entries' codes based on a filter criteria.

**Comment by bmward80** [comment]
Hi @IRodriguezSC,

Yes, you can add values to your multiple choice fields. In the field's `ValueAsString` property, you can just add choices separated by the newline character: `n

```powershell
$choices = "Choice 1`nChoice 2`nChoice 3"
```

Let me know if this helps or if you need any further clarification.

Brandon

**Reply by longj08** [reply]
We tried to add the below values to a multiple choice field in a task with the below codes but the field is not showing the below choices.  It is still showing the default values as the choices.   Any idea?

$choices = "Choice 1`nChoice 2`nChoice 3"
$task.forms.sections.fields.ValueAsString = $choices
Set-Task -Entry $task | Out-Null

Thanks,
Jack

**Reply by bmward80** [reply, marked as answer]
I see, thanks for that, Jack. I think I misunderstood the original question to mean setting the _values_ dynamically and not the actual choices that can be selected in the UI. When a task gets created, those choices are actually derived from the form the task uses, so it's not actually possible to change the choices available to be selected from the task, since the property that holds that info for multiple choice fields, the `DefaultValue` field, is read-only.

Sorry for not understanding that at first.
Brandon

**Comment by IRodriguezSC** [comment]
Perfect…thanks for this…will let you know how it goes.
-IR

Ismael Rodriguez | Mgr - Elec Discov/Lit Support
Sullivan & Cromwell LLP
125 Broad Street | New York, NY 10004-2498
+1 212 558 7245 (T)
***@***.******@***.***> | www.sullcrom.com<http://www.sullcrom.com/>

From: Brandon Ward ***@***.***>
Sent: Tuesday, October 22, 2024 2:40 PM
To: sadieblue/agilityblue.workspace-scripts ***@***.***>
Cc: Rodriguez, Ismael ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] Re: [sadieblue/agilityblue.workspace-scripts] Dynamically setting a multiselect field options. (Discussion #34)


Hi @IRodriguezSC [github.com]<https://urldefense.com/v3/__https:/github.com/IRodriguezSC__;!!KhposeJF9Q!XbGfiZ3JREId45nAb6ar3BW3Y5HLWHaSRorYm89W6DusLzDm05bBCAjCG4U-gZLdL0nmiSDx5U-BdoQgrrAbj6Qjvrs$>,

Yes, you can add values to your multiple choice fields. In the field's ValueAsString property, you can just add choices separated by the newline character: `n

$choices = "Choice 1`nChoice 2`nChoice 3"

Let me know if this helps or if you need any further clarification.

Brandon

—
Reply to this email directly, view it on GitHub [github.com]<https://urldefense.com/v3/__https:/github.com/sadieblue/agilityblue.workspace-scripts/discussions/34*discussioncomment-11020744__;Iw!!KhposeJF9Q!XbGfiZ3JREId45nAb6ar3BW3Y5HLWHaSRorYm89W6DusLzDm05bBCAjCG4U-gZLdL0nmiSDx5U-BdoQgrrAbLXQp7-A$>, or unsubscribe [github.com]<https://urldefense.com/v3/__https:/github.com/notifications/unsubscribe-auth/BK4IHLGUZYNULEJV3IVGQWTZ42LYPAVCNFSM6AAAAABQNCVALOVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMBSGA3TINA__;!!KhposeJF9Q!XbGfiZ3JREId45nAb6ar3BW3Y5HLWHaSRorYm89W6DusLzDm05bBCAjCG4U-gZLdL0nmiSDx5U-BdoQgrrAbuZWQC94$>.
You are receiving this because you were mentioned.Message ID: ***@***.******@***.***>>

________________________________
**This is an external message from: ***@***.******@***.***> **

________________________________
This e-mail is sent by a law firm and contains information that may be privileged and confidential. If you are not the intended recipient, please delete the e-mail and notify us immediately.
